### PR TITLE
SEARCH-1003 Fix a couple of issues raised by Sonar.

### PR DESCRIPTION
**Pull request by tpage-alfresco**
Replace some code that throws a NullPointerException with code to explicitly throw
an AlfrescoRuntimeException if the core name is not passed when creating the
transaction report.

Simplify the logic when getting the first live doc for a search term by not reusing
the postings iterator (nb. It wasn't being reused anyway as we always passed null).

### CODE REVIEW

File: alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java [hunk 1]
```diff
@@ -783,27 +783,16 @@ private void actionTXREPORT(SolrQueryResponse rsp, SolrParams params, String cna
...
-            Long txid = Long.valueOf(params.get(ARG_TXID));
-            NamedList<Object> report = new SimpleOrderedMap<Object>();
-            report.add(cname, buildTxReport(getTrackerRegistry(), informationServers.get(cname), cname, tracker, txid));
-            rsp.add("report", report);
-        }
-        else
-        {
-            Long txid = Long.valueOf(params.get(ARG_TXID));
-            NamedList<Object> report = new SimpleOrderedMap<Object>();
-            for (String coreName : trackerRegistry.getCoreNames())
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java]: The only way this might not throw an exception is if there are no cores in the registry.  I've assumed that this won't happen, but it's worth reviewing.

File: alfresco-search/src/main/java/org/apache/solr/handler/component/AlfrescoLukeRequestHandler.java [hunk 2]
```diff
@@ -461,19 +461,14 @@ private static String getFieldFlags(SchemaField f) {
...
 		TermsEnum termsEnum = terms.iterator();
-		BytesRef text;
 		// Deal with the chance that the first bunch of terms are in deleted
 		// documents. Is there a better way?
-		for (int idx = 0; idx < 1000 && postingsEnum == null; ++idx) {
-			text = termsEnum.next();
-			if (text == null) { // Ran off the end of the terms enum without
-								// finding any live docs with that field in
-								// them.
+		for (int idx = 0; idx < 1000; ++idx) {
```
Comment (tpage-alfresco) [inline alfresco-search/src/main/java/org/apache/solr/handler/component/AlfrescoLukeRequestHandler.java]: I made a mistake here. I assumed that postingsEnum must be null, but actually after the first iteration, postingsEnum must *not* be null.  I should fix this by removing the for loop.

File: alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java (hunk 1 above)
Comment (michaelsuzukisagi) [reply alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java]: This would change the behaviour of the report, not having cname would return a report for all the cores acl txns. The proposed change will now throw "No cname parameter set"

Comment (tpage-alfresco) [reply alfresco-search/src/main/java/org/alfresco/solr/AlfrescoCoreAdminHandler.java]: Not having cname set would previously throw a NullPointerException in trackerRegistry.getTrackerForCore (which will then be wrapped by a SolrException). With the new code we will get an AlfrescoRuntimeException wrapped by a SolrException.

The only edge case is that previously this method would do nothing if there are no cores in the registry and null is passed in. Now it will throw an exception.

I'll add some unit tests and check them before and after my changes.  I don't plan to add a unit test for the edge case I've described as I don't think it's a reasonable use of the transaction reporting.

**Comment by michaelsuzukisagi** [comment]
LGTM
